### PR TITLE
OSOE-795:   Upgrade to latest OC preview to test System.Text.Json

### LINKED ISSUE #692: Upgrade to latest OC preview to test System.Text.Json (OSOE-795)
After https://github.com/OrchardCMS/OrchardCore/pull/14572, OC now uses `System.Text.Json` instead of Json.NET. So, let's upgrade to this preview version and test if anything is broken.

- [x] Open a `task/system-text-json-migration` branch from `dev` and start your issue branch from there. After this issue is done, we won't yet merge this to `dev`, waiting for that for OC v2.0.
- [x] Upgrade OC as usual.
- [x] Change all of our code that uses Json.NET to use STJ instead.
- [x] The https://github.com/OrchardCMS/OrchardCore/pull/13181 PR [changed the registration order of the authentication middleware](https://github.com/OrchardCMS/OrchardCore/pull/13181/files#diff-79fbaf725f0ce29f3982bdf05070824259b3cb3e0f5a523324151423cadef58cL520). This might break [`BlockMediaThemeTemplateDirectAccessMiddleware`](https://github.com/Lombiq/Hosting-Media-Theme/blob/dev/Lombiq.Hosting.MediaTheme.Bridge/Middlewares/BlockMediaThemeTemplateDirectAccessMiddleware.cs). Be sure it's not broken.
- [x] Issues like https://github.com/OrchardCMS/OrchardCore/issues/15628 can now be made apparent, even with SQLite, with the new `EnableThreadSafetyChecks` YesSql config. Please enable those for UI tests, see https://github.com/OrchardCMS/OrchardCore/pull/15961.
- [x] For every general OC issue we find, open an OC issue.
  - https://github.com/OrchardCMS/OrchardCore/issues/15397
  - https://github.com/OrchardCMS/OrchardCore/issues/15451
  - https://github.com/OrchardCMS/OrchardCore/issues/15497
  - https://github.com/OrchardCMS/OrchardCore/issues/15505
  - https://github.com/OrchardCMS/OrchardCore/issues/15533
  - https://github.com/OrchardCMS/OrchardCore/issues/15628
  - https://github.com/OrchardCMS/OrchardCore/issues/15975

[Jira issue](https://lombiq.atlassian.net/browse/OSOE-795)

### PR DESCRIPTION
**Pull request by sarahelsaig**
[OSOE-795](https://lombiq.atlassian.net/browse/OSOE-795)
Fixes #692

[OSOE-795]: https://lombiq.atlassian.net/browse/OSOE-795?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -48,7 +48,7 @@ jobs:
 
   spelling:
     name: Spelling
-    uses: Lombiq/GitHub-Actions/.github/workflows/spelling.yml@dev
+    uses: Lombiq/GitHub-Actions/.github/workflows/spelling.yml@issue/OSOE-795
```
Comment (dministro) [inline .github/workflows/build-and-test.yml]: There is no PR for the branch `issue/OSOE-795` under LGHA

File: .github/workflows/build-and-test.yml [hunk 2]
```diff
@@ -15,7 +15,7 @@ jobs:
     with:
       parent-job-name: "root-solution-larger-runners"
       machine-types: "['buildjet-4vcpu-ubuntu-2204']"
-      timeout-minutes: 30
+      timeout-minutes: 50
```
Comment (dministro) [inline .github/workflows/build-and-test.yml]: Why this increased timeout is needed?

File: tools/Lombiq.GitHub.Actions
Comment (dministro) [inline tools/Lombiq.GitHub.Actions:1]: There is no PR for the branch `issue/OSOE-795` under LGHA

File: src/Modules/Lombiq.Hosting.Azure.ApplicationInsights
Comment (dministro) [inline src/Modules/Lombiq.Hosting.Azure.ApplicationInsights:1]: There is no PR for the branch `issue/OSOE-795`

**Review by dministro — CHANGES_REQUESTED** [review]
Is `task/system-text-json-migration` the correct target branch?

As I see under other repositories, the answer is yes.

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (sarahelsaig) [reply .github/workflows/build-and-test.yml]: Right, and PR for Orchard-Azure-Application-Insights was also missing. I've created both.

File: .github/workflows/build-and-test.yml (hunk 2 above)
Comment (sarahelsaig) [reply .github/workflows/build-and-test.yml]: I think it was accidental.

**Review by dministro — APPROVED** [review]

**Review by dministro — APPROVED** [review]

**Pull request merged**